syntax = "proto3";

service RegistrationService {
  rpc RegisterClient (ClientInfo) returns (ClientRegistrationResponse) {}
}

service DirectoryService {
  rpc GetDirectory (ClientInfo) returns (ClientDirectory) {}
}

service RoutingService {
  rpc SendMessage (MessageRequest) returns (MessageResponse) {}
}

message ClientInfo {
  string name = 1;
  int32 id = 2;
}

message ClientRegistrationResponse {
  int32 id = 1;
}

message ClientDirectory {
  repeated ClientInfo clients = 1;
}

message MessageRequest {
  int32 sender_id = 1;
  int32 receiver_id = 2;
  string message = 3;
}

message MessageResponse {
  bool success = 1;
}

